point_pattern.py

```diff
@@ -102,24 +117,41 @@ def mean_center(points):
 def average_nearest_neighbor_distance(points):
     """
     Given a set of points, compute the average nearest neighbor.
-
     Parameters
     ----------
     points : list
              A list of points in the form (x,y)
-
     Returns
     -------
     mean_d : float
              Average nearest neighbor distance
-
     References
     ----------
     Clark and Evan (1954 Distance to Nearest Neighbor as a
      Measure of Spatial Relationships in Populations. Ecology. 35(4)
      p. 445-453.
     """
-    mean_d = 0
+    sum_nn_dis = 0
+    
+    for point_1 in points:
+        first = True
+        for point_2 in points:
+            if point_1 == point_2:
+                continue
```

Comment:
Nice use of continue.  Check out `enumerate()` as well.  This logic works this week, but fails if the points are coincident.  Since you know that the lists are in the same order, you could check to see if the positional indices are the same.